### PR TITLE
feat(medusa): Categories -  Adds indexes + remove soft delete

### PR DESCRIPTION
**Pull request by riqwan**
What:
- Removes soft delete: This isn't really a needed feature here and brings about some issues undeleting that isn't currently addressed. 
- Adds commonly scoped attributes to indexes

### CODE REVIEW

File: packages/medusa/src/migrations/1679921553298-product_category_indexes.ts
Comment (adrien2p) [inline packages/medusa/src/migrations/1679921553298-product_category_indexes.ts:1]: question: It feels like we are indexing all the columns of the entity. One thing is that doing such thing can end up being the same as not indexing as all the data will end up in the indexes. any particular reason why we don't create an index for example on all non internal and active categories?

File: packages/medusa/src/migrations/1679926917252-remove_deleted_at.ts
Comment (adrien2p) [inline packages/medusa/src/migrations/1679926917252-remove_deleted_at.ts:3]: todo: rename, the naming is mistaking the reader :)

**Review by olivermrbl — APPROVED** [review]
LGTM - pending @adrien2p's approval 💯

File: packages/medusa/src/migrations/1679950221063-category-remove-soft-delete.ts
Comment (adrien2p) [inline packages/medusa/src/migrations/1679950221063-category-remove-soft-delete.ts:1]: Todo: depending on the decision it is a todo or not :)
Since some user might have tested the categories already and maybe deleted some of them, and that we are deleting the deleted_at, should we also remove all tuple where delete at is not null before dropping the column. In order to avoid the previously deleted categories to pop up again?

**Review by adrien2p — CHANGES_REQUESTED** [review]

**Review by adrien2p — APPROVED** [review]
LGTM 💪

**Comment by olivermrbl** [comment]
Seems like there's an issue with the integration tests. The error is a bit strange. Do yours pass locally @riqwan ?

**Comment by adrien2p** [comment]
> Seems like there's an issue with the integration tests. The error is a bit strange. Does yours pass locally @riqwan?

looks like the variants are not in the right order in the result compared to the expected order

**Comment by riqwan** [comment]
Weird, that has nothing to do with this PR. But its passing locally for me.

File: packages/medusa/src/migrations/1679950221063-category-remove-soft-delete.ts
Comment (riqwan) [reply packages/medusa/src/migrations/1679950221063-category-remove-soft-delete.ts:1]: good catch! <3

**Comment by olivermrbl** [comment]
@riqwan we're good now - will publish a patch of our RC later for testing

**Pull request merged**
